### PR TITLE
Export help window to workbench

### LINKED ISSUE #24709: Export HelpWindow to the Workbench
The [`HelpWindow`](https://github.com/mantidproject/mantid/blob/master/qt/widgets/common/inc/MantidQtWidgets/Common/HelpWindow.h) class is used to provide the offline help for MantidPlot. The same class should be able to be used for the workbench Python code. It needs to be accessible from Python so will require exporting and having wrapper modules provided. Once exposed to Python the current python references that access the `HelpWindow` should be updated to use `mantidqt`.

The `HelpWindow` needs to be included in the Qt5 build of the `MantidWidgetsCommon` library and when this is done the `?` button on the algorithm dialogs should work. This needs to be verified.

### PR DESCRIPTION
**Pull request by ghost**
**Description of work.**
Enable offline help in Workbench.

For Windows this involved having CMake move QtWebEngine resource files to the package build and embedding a `qt.conf` file in Workbench's `resources.py` to direct Qt to those files. Embedding was necessary to avoid pointing MantidPlot to the wrong places.

**To test:**
- Open Workbench and open the help window (either via the help menu or using the F1 shortcut).
- Make sure:
    - help pages are displayed correctly
    - http links open in your web browser
    - links direct you to the correct page
    - back and forward buttons update/function correctly
- Check the help window opens the correct page when clicking the help button in an algorithm run dialog
- Check this on a developer and package build.
- Check the help window still works in MantidPlot

Note: If you're on Windows and do not have Visual Studio 2015 you may need to implement the fix described in #25039 to build the package.

Fixes #24709. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
I can confirm this works on macOS with a package.
I will test it on Ubuntu.

**Review by martyngigg — CHANGES_REQUESTED** [review]
All the help window stuff gets displayed on Ubuntu 16.04 when I build a package but I seem to get a segfault every time I quit:
```
Thread 1 "python" received signal SIGSEGV, Segmentation fault.
0x00007fffe6a18261 in QSqlDatabase::close() () from /usr/lib/x86_64-linux-gnu/libQt5Sql.so.5
(gdb) bt
#0  0x00007fffe6a18261 in QSqlDatabase::close() () from /usr/lib/x86_64-linux-gnu/libQt5Sql.so.5
#1  0x00007fffe6a197c9 in QSqlDatabase::~QSqlDatabase() () from /usr/lib/x86_64-linux-gnu/libQt5Sql.so.5
#2  0x00007fffe6a1b7ed in ?? () from /usr/lib/x86_64-linux-gnu/libQt5Sql.so.5
#3  0x00007ffff4d99bf9 in QHashData::free_helper(void (*)(QHashData::Node*)) () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#4  0x00007fffe6a17f33 in ?? () from /usr/lib/x86_64-linux-gnu/libQt5Sql.so.5
#5  0x00007ffff75dcff8 in __run_exit_handlers (status=0, listp=0x7ffff79675f8 <__exit_funcs>, run_list_atexit=run_list_atexit@entry=true) at exit.c:82
#6  0x00007ffff75dd045 in __GI_exit (status=<optimised out>) at exit.c:104
#7  0x000000000051ea3f in Py_Exit (sts=sts@entry=0) at ../Python/pythonrun.c:1785
#8  0x000000000051bfd7 in handle_system_exit () at ../Python/pythonrun.c:1151
#9  0x000000000051b8fd in PyErr_PrintEx () at ../Python/pythonrun.c:1161
#10 0x0000000000430c25 in PyErr_Print () at ../Python/pythonrun.c:1064
#11 PyRun_SimpleFileExFlags (fp=<optimised out>, filename=<optimised out>, closeit=<optimised out>, flags=<optimised out>) at ../Python/pythonrun.c:952
#12 0x00000000004938ce in Py_Main () at ../Modules/main.c:640
#13 0x00007ffff75c3830 in __libc_start_main (main=0x493370 <main>, argc=2, argv=0x7fffffffd4a8, init=<optimised out>, fini=<optimised out>, rtld_fini=<optimised out>, stack_end=0x7fffffffd498) at ../csu/libc-start.c:291
#14 0x0000000000493299 in _start ()
(gdb) 
```
Have you seen this?

File: buildconfig/CMake/WindowsNSISQt5.cmake [hunk 1]
```diff
@@ -8,6 +8,17 @@ set ( QT5_DIST_DLLS
     Qt5PrintSupport.dll
     Qt5Svg.dll
     Qt5OpenGL.dll
+    Qt5Help.dll
+    Qt5Network.dll
+    Qt5Positioning.dll
```
Comment (martyngigg) [inline buildconfig/CMake/WindowsNSISQt5.cmake]: Do we need things like Positioning, Qml, QuickWidgets?

Comment (hsautessella) [reply buildconfig/CMake/WindowsNSISQt5.cmake]: I haven't seen any errors like that, but I'll build the package on my 18.04 VM and see if I can replicate it. 

I haven't tried without those DLLs, but everything added were things that the dependency walker threw up as missing. But I'll have a play around and see what can be taken out.

File: qt/applications/workbench/workbench/app/mainwindow.py [hunk 2]
```diff
@@ -12,6 +12,9 @@
 """
 from __future__ import (absolute_import, division, print_function, unicode_literals)
 
+# Importing resources loads the data in
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/app/mainwindow.py]: Could you add a comment as to indicate why this has to be done so early? I can imagine someone will try to tidy up and move this.

File: qt/python/mantidqt/_common.sip [hunk 3]
```diff
@@ -224,6 +224,14 @@ public:
   GenericDialog(QWidget *parent = nullptr);
 };
 
+class UserSubWindow : QMainWindow
```
Comment (martyngigg) [inline qt/python/mantidqt/_common.sip]: Was this required for exporting the help? It should be possible to leave out the `createSubWindow` definition.

**Review by martyngigg — CHANGES_REQUESTED** [review]
A few more minor comments.

File: buildconfig/CMake/WindowsNSISQt5.cmake (hunk 1 above)
Comment (martyngigg) [reply buildconfig/CMake/WindowsNSISQt5.cmake]: Okay. If dependency walker says that then its probably correct. I was just suprised it needs to much stuff as we're not using Qml etc.

For the segfault I wonder if the SQL connection is being closed down too late in the python exit process. There is a `shutdown` method on the `MantidHelpWindow`. Perhaps we need to call that in the `MainWindow.closeEvent`?

**Review by martyngigg — DISMISSED** [review]

**Review by martyngigg — APPROVED** [review]

**Pull request merged**